File: refractive_shift.c
#include <stdio.h>
#include <math.h>

#define PI 3.14159265

void var_input(char* str, float* f) {
  printf("%s", str);
  scanf("%f", f);
}

int main(int argc, char** argv) {
  float a, d, n1, n2, x;
  var_input("Incident angle : ", &a);
  var_input("Glass thicc    : ", &d);
  var_input("Index 1        : ", &n1);
  var_input("Index 2        : ", &n2);
  x = ((d*(cos(a*(PI/180)+asin((n1*cos(a*(PI/180)))/(n2)))))/(cos(asin((n1*cos(a*(PI/180)))/(n2)))));
  printf("Refractive shift : %3f\n", x);
  return 0;
}

File: src/samples/simulations/refractive_shift/refractive_shift.py
import matplotlib.pyplot as plt
import math

n1 = 1
n2 = 1.5
d = 2.7
PI = 3.14159265
a = 0
angles = []
for a in range(0, 91):
    x = ((d*(math.cos(a*(PI/180)+math.asin((n1*math.cos(a*(PI/180)))/(n2)))))/(math.cos(math.asin((n1*math.cos(a*(PI/180)))/(n2)))))
    angles.append(x)
    print("{} degrees is : {}".format(a, x))

plt.plot(range(len(angles)), angles)
plt.ylabel('Refractive shift')
plt.xlabel('Angle in degrees')
plt.title('Relationship between horizontal incident angle and refractive shift with glass thickness : {}'.format(d))
plt.show()
